articles/112_generated_interfaces_cpp.md

```diff
@@ -104,6 +104,13 @@ All other constants are declared as `static const` members in the struct and the
 ### Constructors
 
 The *default constructor* initializes all members with their default value.
+In some cases this may not be desirable, since these fields will often be immediately overwritten with user-provided values.
+Therefore, the constructor takes an optional directive of type `enum rcl_msg_init_type` to control how initialization is done:
+
+- `MSG_INIT_DEFAULT_INITIALIZE_ALL` - Default initialize all members; any fields that have default values assigned to individual members will be set to the default values, all other values will be assigned to C++ defaults (generally 0 or the empty string).  Equivalent to not passing any argument to the constructor
+- `MSG_INIT_DEFAULT_INITIALIZE_NONE` - Don't initialize any members; it is the user's responsibility to ensure that all fields get initialized with some value, otherwise undefined behavior results
+- `MSG_INIT_ZERO_INITIALIZE_ALL` - Zero initialize all members; this differs from `MSG_INIT_DEFAULT_INITIALIZE_ALL` in that all members will be set to their C++ defaults (generally 0 or the empty string), and default values from the message definition will be ignored
+
```

Comment:
The four options we discussed in the meeting are:

* `SKIP_INITIALIZE` - Does neither set members which have default values defined in the IDL nor set member to a type-specific default value.
  * This is used for maximum performance if the user sets all members on his own anyway.
* `INITIALIZE_DEFAULTS_ONLY` - Only initializes the members which have default values defined in the IDL. The other member are not initialized and are expected to be set by user land code.
  * This is the "minimal" initialization that ensures that existing code continues to have correctly initialized member when a new field with a default value is added later to the IDL.
* `INITIALIZE_ALL` - Initializes the members which have default values defined in the IDL. All other members are initialized with their type-specific default value.
  * This is the safest options and should be the "default".
* `ZERO_INITIALIZE` - Initializes all members with their type-specific default value not considering any default values specified in the IDL.
  * This could be used when the user doesn't want the overhead of potentially complex / large default values (since they are overridden anyway) but still wants to ensure that all variables are properly initialized.